### DISCUSSION TITLE
Why PersistentTopicInternalStats don't provide getter/setter method?

### DISCUSSION THREAD

**Discussion by crossoverJie** [apache/pulsar#18481] [Q&A] [answered]
https://github.com/apache/pulsar/blob/master/pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/policies/data/PersistentTopicInternalStats.java

https://github.com/apache/pulsar/blob/master/pulsar-client-admin-api/src/main/java/org/apache/pulsar/common/partition/PartitionedTopicMetadata.java

When I am using the admin API, I find that there is no getter/setter method for the returned data, and I would like to know the reason for this design.

The first impression is that it is not Java-like.

**Comment by michaeljmarshall** [comment, marked as answer]
Seems like an oversight to me. There could definitely be getter methods for the fields in those classes, but I don't think we need setter methods. Similarly, `org.apache.pulsar.common.policies.data.PartitionedTopicStats` only has getters.

**Reply by crossoverJie** [reply]
Thank you for your answer, I think I can help fix this.
